### DISCUSSION TITLE
INSERT INTO ... ON CONFLICT ... query doesnt seem to update the "updated" column

### DISCUSSION THREAD

**Discussion by lordvcs** [pocketbase/pocketbase#4644] [Q&A] [answered]
I have a `otps` table with a `user` single relation with unique index on it

`app.Dao().DB().NewQuery("INSERT INTO otps (otp, user) VALUES ({:otp}, {:user}) ON CONFLICT(user) DO UPDATE SET otp = {:otp}").Bind(dbx.Params{"otp": "9992", "user": user.Id}).Execute()`
				
For this query, I expected the "updated" column to have its value updated automatically with the current datetime whenever the same user id is used, but it never changes no matter how many times I run the query.
Is this expected behaviour?
I tried `updated=datetime('now') `, and that is working fine

**Comment by ganigeorgiev** [comment, marked as answer]
If you are running raw insert/update SQL queries you have to set its value manually.

The `updated` fields is updated automatically only when the query is executed via the app Dao model helpers.

**Reply by ganigeorgiev** [reply]
Side-note: if you are going to update it manually prefer using `strftime('%Y-%m-%d %H:%M:%fZ')` instead of `datetime('now')` as the first one will produce the current full `date` field value format (_both technically will work as the value will be normalized when unmarshaling the db result but the latter will differ from the normal stored values and may cause issues when performing `WHERE` searches on the field_).

**Reply by mlusetti** [reply]
I've had this similar situation and can confirm what Gani said about strftime / datetime usage.
I've had bad time troubleshooting where issues were.

**Reply by lordvcs** [reply]
@ganigeorgiev maybe we should add this info to the docs?

btw, thanks for all ur work
